packages/typespec-client-generator-core/src/http.ts

```diff
@@ -180,20 +183,16 @@ function getSdkHttpParameters(
   // we add correspondingMethodParams after we create the type, since we need the info on the type
   const correspondingMethodParams: (SdkMethodParameter | SdkModelPropertyType)[] = [];
   if (tspBody) {
-    if (tspBody.bodyKind === "file") {
-      // file body is not supported yet
-      diagnostics.add(
-        createDiagnostic({
-          code: "unsupported-http-file-body",
-          target: tspBody.property ?? tspBody.type,
-        }),
-      );
-      return diagnostics.wrap(retval);
-    }
     if (tspBody.property && !isNeverOrVoidType(tspBody.property.type)) {
       const bodyParam = diagnostics.pipe(
         getSdkHttpParameter(context, tspBody.property, httpOperation.operation, undefined, "body"),
       );
+      if (tspBody.bodyKind === "file" && bodyParam.kind === "body") {
+        if (!bodyParam.serializationOptions) {
+          bodyParam.serializationOptions = {};
+        }
```

Comment:
nit: I think this could also be expressed as
```suggestion
        bodyParam.serializationOptions = bodyParam.serializationOptions ??  {};
```